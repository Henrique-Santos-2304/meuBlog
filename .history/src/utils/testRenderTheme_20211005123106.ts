import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
