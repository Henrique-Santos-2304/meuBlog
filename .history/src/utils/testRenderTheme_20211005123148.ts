import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
