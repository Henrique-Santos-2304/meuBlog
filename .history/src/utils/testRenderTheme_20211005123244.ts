import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
