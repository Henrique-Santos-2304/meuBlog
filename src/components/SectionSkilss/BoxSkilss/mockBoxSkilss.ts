import { mockSlides } from "components/globalComponents/SliderPhoto/mockSlides";

export const mockBoxSkill = {
  data: mockSlides.images[0],
  marginBox: "1rem",
};
