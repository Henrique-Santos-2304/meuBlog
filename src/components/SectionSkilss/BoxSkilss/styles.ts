import styled from "styled-components";
import Image from "next/image";

type marginProps = {
  marginBox: string;
};
export const ContainerSkilss = styled.div<marginProps>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background: rgba(0, 0, 0, 0.55);
  margin: ${({ marginBox }) => marginBox};
  padding: 1rem 0.5rem;
  border-radius: 1rem;
  position: relative;

  &::before {
    content: "";
    position: absolute;
    width: 0.3rem;
    height: 70%;
    top: 50%;
    right: -1.5rem;
    transform: translateY(-50%);
    background: ${({ theme }) => theme.colors.effectsPrimary};
  }
`;
export const ContentImage = styled.div`
  width: 5rem;
  height: 4rem;
  position: relative;
`;
export const ImageSkilss = styled(Image)``;

export const TitleSkilss = styled.h3`
  margin-top: 1.3rem;
  font-size: ${({ theme }) => theme.font.sizes.small};
  font-weight: 300;
  text-align: center;
  color: ${({ theme }) => theme.colors.primary};
`;
