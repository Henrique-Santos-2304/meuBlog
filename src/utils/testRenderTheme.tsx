import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
