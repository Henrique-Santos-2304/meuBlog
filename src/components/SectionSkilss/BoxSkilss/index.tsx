import * as S from "./styles";

type boxSkilssProps = { img: string; title: string; marginBox: string };

const BoxSkilss = ({ img, title, marginBox }: boxSkilssProps) => (
  <S.ContainerSkilss
    aria-label="Linguagens,Frameworks ou bibliotecas dominadas"
    marginBox={marginBox}
  >
    <S.ContentImage>
      <S.ImageSkilss src={img} layout="fill" />
    </S.ContentImage>
    <S.TitleSkilss>{title}</S.TitleSkilss>
  </S.ContainerSkilss>
);

export default BoxSkilss;
