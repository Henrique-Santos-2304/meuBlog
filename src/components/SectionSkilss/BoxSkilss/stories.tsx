import { Story, Meta } from "@storybook/react";
import BoxSkilss from ".";

export default {
  title: "BoxSkilss",
  component: BoxSkilss,
} as Meta;

export const Basic: Story = (args) => <BoxSkilss {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <BoxSkilss {...args} />;
