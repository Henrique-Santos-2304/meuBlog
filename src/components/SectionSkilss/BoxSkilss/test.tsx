import BoxSkilss from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

const mock = {
  img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634679113/rtl_i3scd8_11f5f28c30.svg",
  title: "Titulo da imagem",
  marginBox: "1rem",
};
describe("<BoxSkilss />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<BoxSkilss {...mock} />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<BoxSkilss {...mock} />);
    const title = screen.getByRole("heading", { name: /Titulo da imagem/i });
    const image = screen.getByRole("img");
    expect(title).toBeInTheDocument();
    expect(image).toBeInTheDocument();
  });
});
