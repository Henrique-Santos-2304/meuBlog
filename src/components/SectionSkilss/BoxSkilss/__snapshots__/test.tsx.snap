// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxSkilss /> should render  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0,0,0,0.55);
  margin: 1rem;
  padding: 2rem;
  border-radius: 1rem;
  position: relative;
}

.c0::before {
  content: "";
  position: absolute;
  width: 0.3rem;
  height: 70%;
  top: 50%;
  right: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  background: #b7f600;
}

.c1 {
  width: 5rem;
  height: 5rem;
  position: relative;
}

.c2 {
  margin-top: 1.3rem;
  font-size: 1.4rem;
  font-weight: 300;
  text-align: center;
  color: #f8f8f8;
}

<div>
  <div
    aria-label="Linguagens,Frameworks ou bibliotecas dominadas"
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          class=""
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </div>
    </div>
    <h3
      class="c2"
    >
      Titulo da imagem
    </h3>
  </div>
</div>
`;
